docs/articles/documentation/using-testcafe/configuration-file.md

```diff
@@ -0,0 +1,764 @@
+
+*CLI*: [--speed](command-line-interface.md#--speed-factor)  
+*API*: [runner.run({ speed })](programming-interface/runner.md#run)
+
+## port1, port2
+
+Specifies custom port numbers TestCafe uses to perform testing. The number range is [0-65535].
+
+```json
+{
+    "port1": 12345,
+    "port2": 54321
+}
+```
+
+TestCafe automatically selects ports if ports are not specified.
+
+*CLI*: [--ports](command-line-interface.md#--ports-port1port2)  
+*API*: [createTestCafe](programming-interface/createtestcafe.md)
+
+## hostname
+
+Specifies your computer's hostname. It is used when you run tests in remote browsers.
+
+```json
+{
+    "hostname": "host.mycorp.com"
+}
+```
+
+If the hostname is not specified, TestCafe uses the operating system's hostname or the current machine's network IP address.
+
+*CLI*: [--hostname](command-line-interface.md#--hostname-name)  
+*API*: [createTestCafe](programming-interface/createtestcafe.md)
+
+## proxy
+
+Specifies the proxy server used in your local network to access the Internet.
+
+```json
+{
+    "proxy": "proxy.corp.mycompany.com"
+}
+```
+
+```json
+{
+    "proxy": "172.0.10.10:8080"
+}
+```
+
+You can also specify authentication credentials with the proxy host.
+
+```json
+{
+    "proxy": "username:password@proxy.mycorp.com"
+}
+```
+
+*CLI*: [--proxy](command-line-interface.md#--proxy-host)  
+*API*: [runner.useProxy](programming-interface/runner.md#useproxy)
+
+## proxyBypass
+
+Specifies the resources accessed bypassing the proxy server.
+
+```json
+{
+    "proxyBypass": "*.mycompany.com"
+}
+```
+
+```json
+{
+    "proxyBypass": ["localhost:8080", "internal-resource.corp.mycompany.com"]
+}
+```
+
+See the [--proxy-bypass](command-line-interface.md#--proxy-bypass-rules) command line parameter for details.
+
+*CLI*: [--proxy-bypass](command-line-interface.md#--proxy-bypass-rules)  
+*API*: [runner.useProxy](programming-interface/runner.md#useproxy)
+
+## ssl
+
+Provides options that allow you to establish an HTTPS connection between the client browser and the TestCafe server.
+
+```json
+{
+    "ssl": {
+        "pfx": "path/to/file.pfx",
+        "rejectUnauthorized": true
+    }
+}
+```
+
+See the [--ssl](command-line-interface.md#--ssl-options) command line parameter for details.
+
+*CLI*: [--ssl](command-line-interface.md#--ssl-options)  
+*API*: [createTestCafe](programming-interface/createtestcafe.md)
+
+## developmentMode
+
+Enables mechanisms to log and diagnose errors. You should enable this option if you are going to contact TestCafe Support to report an issue.
+
+```json
+{
+    "developmentMode": true
+}
+```
+
+*CLI*: [--dev](command-line-interface.md#--dev)  
+*API*: [createTestCafe](programming-interface/createtestcafe.md)
+
+## qrCode
+
+If you launch TestCafe from the console, outputs a QR-code that represents URLs used to connect the remote browsers.
```

Comment:
outputs?

Reply:
What 'outputs'?

Reply:
Oh, I meant a verb.

*If you launch TestCafe from the console, this option outputs a QR-code that represents URLs used to connect the remote browsers.*